StortSpelProjekt/Engine/src/Renderer/Renderer.cpp

```diff
@@ -2262,3 +2248,32 @@ void Renderer::toggleFullscreen(WindowChange* evnt)
 		}
 	}
 }
+
+SwapChain* Renderer::getSwapChain()
+{
+	return m_pSwapChain;
+}
+
+void Renderer::submitTextToGPU(Text* text, TextManager* tm)
+{
+	// Submit to GPU
+	const void* data = static_cast<const void*>(text->m_TextVertexVec.data());
+
+	// Vertices
+	Resource* uploadR = text->m_pUploadResourceVertices;
+	Resource* defaultR = text->m_pDefaultResourceVertices;
+	CopyOnDemandTask* codt = static_cast<CopyOnDemandTask*>(m_CopyTasks[COPY_TASK_TYPE::COPY_ON_DEMAND]);
+	codt->Submit(&std::make_tuple(uploadR, defaultR, data));
+
+	AssetLoader* al = AssetLoader::Get();
+	std::wstring fontPath = al->GetFontPath();
+	std::wstring path = fontPath + text->GetFont()->GetName() + L".fnt";
+	bool isTextureOnGpu = al->m_LoadedFonts[path].first;
```

Comment:
I would create a public function in AssetLoader "IsFontLoaded()".

Reply:
Just a object oriented code visibility thing you then "ask" assetloader if it's loaded and you don't "peek" yourself.